cmake/modules/verify-toolchain.cmake

```diff
@@ -76,7 +76,7 @@ if(("zephyr" STREQUAL ${ZEPHYR_TOOLCHAIN_VARIANT}) OR
     # To support Zephyr SDK tools (DTC, and other tools) with 3rd party toolchains
     # then we keep track of current toolchain variant.
     set(ZEPHYR_CURRENT_TOOLCHAIN_VARIANT ${ZEPHYR_TOOLCHAIN_VARIANT})
-    find_package(Zephyr-sdk ${TOOLCHAIN_ZEPHYR_MINIMUM_REQUIRED_VERSION} REQUIRED QUIET HINTS $ENV{ZEPHYR_SDK_INSTALL_DIR})
+    find_package(Zephyr-sdk ${TOOLCHAIN_ZEPHYR_MINIMUM_REQUIRED_VERSION} REQUIRED QUIET HINTS ${ZEPHYR_SDK_INSTALL_DIR})
```

Comment:
no, this should not change.

The environment is how users are used to work with ZEPHYR_SDK_INSTALL_DIR, CMake itself offers the possibility of using `<packagename>_ROOT` for seaching, which can be set as CMake variable.

Ref: https://cmake.org/cmake/help/latest/command/find_package.html#config-mode-search-procedure
> 1. New in version 3.12: Search paths specified in the `<PackageName>_ROOT` CMake variable and the `<PackageName>_ROOT` environment variable, where <PackageName> is the package to be found.

Reply:
My bad, I didn't notice we are already mapping the environment setting into a CMake variable (if not already set):
https://github.com/zephyrproject-rtos/zephyr/blob/cd7c44a152bc4469ad7da9e38d58df46655d460d/cmake/modules/verify-toolchain.cmake#L28-L29

so agreed, better to use the CMake var on this.
Although we should probably consider better alignment with CMake natively supported `find_package()`, but that is outside scope of this PR.